testbed/tests/syslog_integration_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package tests // import "github.com/open-telemetry/opentelemetry-collector-contrib/testbed/tests"
+
+import (
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/datareceivers"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/testbed"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/otelcol"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+type expectedDataType struct {
+	message        string
+	severityNumber plog.SeverityNumber
+	severityText   string
+	timestamp      pcommon.Timestamp
+	attributes     map[string]interface{}
+}
+
+func TestSyslogComplementaryRFC5424(t *testing.T) {
```

Comment:
Both these tests randomly fail on my machine. We have investigated this with @sumo-drosiek and we believe we found the cause and the fix for this. I was actually just going to submit this fix in relation to this issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21244. I'm fine with merging this PR as long as the CI tests pass, I will submit the fix in a separate PR.